research-2023/effort-mapping-model/2023-04-28.001.md

```diff
@@ -0,0 +1,321 @@
+
+While OSCAL is attempting to place content into a machine readable format, there is still a person, or group of people, involved in the process for some of the activities where subjective decisions need to be made. Frameworks are authored by different organizations with different perspectives, goals and vocabulary. It seems that the ability to document enough for a machine to be aware of this context would facilitate current methods as a step forward, without being entirely mathematical or free of bias in the mapping itself.
+
+We also have the ability to improve as OSCAL matures, and the constraints of paper documents and spreadsheets fade.
+
+---
+
+### New Assembly: Qualifier
+
+The qualifier is loosely based on the Resource Description Framework (RDF) subject-predicate-object[^rdf] and applies to each map object.  How closely we want to match this, or implement this was beyond the scope of just capturing the essence of an assembly that would allow expression of differences without implying connotations on the context.  For example, expressing a "gap" or "discrepancy" in a tag could invite conflict if this could be interpreted as a framework as less useful or capable than another.
+
+Very rough concept:
+```xml
+<qualifier
+	subject=“source|target|both”
+	has=“requirement|incompatibility”
+	category=“restricted|addressable|blocked”>
+    Description of qualifier...
+</qualifier>
+```
+
+---
+
+### Option: String Marker in Qualifer
+
+Additional attributes could be considered for marking a start and end to the control/statement string to highlight the portion referenced by the qualifier.
+
+```
+	start=“starting position in string”
+	end=“ending position in string”
+```
+
+There are other potential approaches here, and this is likely an "overdone" aspect of our desire to implement just enough to make the model useful as a first release.
+
+---
+
+### New Assembly: Provenance
+
+This assembly applies to the entire mapping and provides clear context on intended use, the author(s) of the mapping, the manner it was achieved, and the state of the mapping.  If we imagine a repository of 200 mapping documents, this provides enough context to traverse all mappings, or query them, in specific ways to eliminate undesired mappings depending on the use case.  For example, one may wish to only use content produced by people, with a complete status.  Another example might allow a CISO to visually explore future or proposed changes to a standard.
+
+Very rough concept:
+```xml
+<provenance
+	responsible-parties=""
+	method=“human|automation”
+	matching=“lexical|logical|semantic|etc”
+	confidence-score=""
```

Comment:
Blocking, important question: I would like clarification here: what do we intend `confidence-score` to be? A percentage, an adjective with a qualitative description?

Reply:
I was (personally envisioning it as a numeric value (0,10), with 10 very confident. BUT, I was thinking each  relationship/map should have this type of score and the field in here to capture (min, max) OR better, the average, because it will better reflect the individual scores.

I have seen NLP models providing confidence score of the content understanding so tools relying on NLP can calculate such score in a consistent way. A human will have to manually provide it, but we all assess pains in the way in doctor's office and mappings are painful 😀

Reply:
This concept comes from the SSSOM reference in the spiral.  

Specifically see: https://mapping-commons.github.io/sssom/confidence/

> A score between 0 and 1 to denote the confidence or probability that the match is correct, where 1 denotes total confidence.

(0 to 1 supports machine interpretation without refitting data/feature engineering.)

Reply:
Thanks for the follow-up comments. There was ample explanation here and a brief by Chris to Michaela and I. I have some recommendations around the structuring of the data element to be added because of data type flexibility, but I am supportive overall. As agreed that can come up in the context of the feature request, code review, and community feedback.